### PR TITLE
feat: Manage-SDK, removeForApp, clear and clearForApp for secure storage

### PR DESCRIPTION
**Pull request by kevinshahfws**

### CODE REVIEW

File: src/openrpc/secure_storage.json [hunk 1]
```diff
@@ -256,6 +256,168 @@
           }
         }
       ]
+    },
+    {
+      "name": "setForApp",
+      "summary": "Set or update a secure data value from distributor for an appId",
```
Comment (jlacivita) [inline src/openrpc/secure_storage.json:262]: It's not necessary going to be called by a distributor. Maybe change this to "for a specific app."

File: src/openrpc/secure_storage.json [hunk 2]
```diff
@@ -256,6 +256,168 @@
...
+          "result": {
+            "name": "defaultResult",
+            "value": null
+          }
+        }
+      ]
+    },
+    {
+      "name": "removeAll",
+      "summary": "Remove all secure data values by distributor for an appId",
```
Comment (jlacivita) [inline src/openrpc/secure_storage.json:380]: Same not on "distributor"

File: src/openrpc/secure_storage.json [hunk 3]
```diff
@@ -256,6 +256,168 @@
...
+          ],
+          "result": {
+            "name": "defaultResult",
+            "value": null
+          }
+        }
+      ]
+    },
+    {
+      "name": "removeAll",
```
Comment (jlacivita) [inline src/openrpc/secure_storage.json:379]: we should call this `removeAllForApp` and have a second method called `removeAll` that doesn't include the `appId` parameter and `uses` the capability.

I also think `clear` might be a better name, but that's subjective.

**Review by jlacivita — CHANGES_REQUESTED** [review]

File: src/openrpc/secure_storage.json [hunk 4]
```diff
@@ -256,6 +256,168 @@
...
       ]
+    },
+    {
+      "name": "setForApp",
+      "summary": "Set or update a secure data value from distributor for an appId",
+      "tags": [
+        {
+          "name": "capabilities",
+          "x-manages": [
+            "xrn:firebolt:capability:storage:secure"
```
[the remark this answers is not in the document]
Comment (kpears201) [reply src/openrpc/secure_storage.json:267]: I do think changing the capability of a method is a breaking change. If we change this for 0.13, then the `storage:secure` capability will technically no longer exist. Not sure what that really means for apps using 0.12 with Ripple that supports 0.13. I guess we would have to map permissions to a capability that no longer exists?

>I'm not sure storage:secure is a good name, as it implies we might have insecure storage, too.
The whole module is called SecureStorage, so shouldn't that statement apply to the module name?
The name SecureStorage is already pretty widely used, might be too late.

[the remark this answers is not in the document]
Comment (jlacivita) [reply src/openrpc/secure_storage.json:267]: ok i’m sold

File: src/openrpc/secure_storage.json [hunk 5]
```diff
@@ -256,6 +256,291 @@
...
+          },
+          "required": true
+        },
+        {
+          "name": "key",
+          "summary": "Key to remove",
+          "schema": {
+            "type": "string"
+          }
+        }
```
Comment (jlacivita) [inline src/openrpc/secure_storage.json:412]: key should be required, right? we don't want people using this api to clear all of the keys in app/scope, that's what clear is for.

**Review by jlacivita — APPROVED** [review]
Looks good, thanks!

**Pull request merged**